dominion/static/js/map_mapit.js

```diff
@@ -0,0 +1,197 @@
+/***
+ * A class that loads geography boundary information from
+ * mapit.code4sa.org.
+ */
+function MapItGeometryLoader() {
+    var self = this;
+    self.mapit_url = MAPIT.url;
+    self.mapit_codetype = MAPIT.code_type;
+    self.mapit_country = MAPIT.map_country;
+    /**
+     * Fetches geometry data for a comparison view and calls the +success+
+     * callback with an object mapping each geo-id to a GeoJSON object.
+     */
+     this.loadGeometryForComparison = function(comparison, success) {
+         // load all country, province, municipality and ward geo data
+         var counter = comparison.geoIDs.length;
+         var featureMap = {};
+         var generation = MAPIT.generations[comparison.geoVersion];
+
+         _.each(comparison.geoIDs, function(geoid) {
+             // eg. province-WC
+             var parts = geoid.split('-'),
+                 level = parts[0],
+                 code = geoid,
+                 url_ = '/code/' + self.mapit_codetype + '/' + code;
+
+             var simplify = MAPIT.level_simplify[level];
+             var mapit_codetype = this.mapit_codetype;
+
+             url_ = url_ + "?generation=" + generation;
+             d3.json(this.mapit_url + url_, function(error, data) {
+               if (error) return console.warn(error);
+               var area = data;
+               var url = '/area/' + area.id + '.geojson?type=' + area.type + "&country=" + area.country+ '&simplify_tolerance='+ simplify;
+               d3.json(self.mapit_url + url, function(error, geojson) {
+                   --counter;
+                   if (error) return console.warn(error);
+                   var features = geojson.features;
+                   feature.properties = {}
+                   feature.properties['area_id'] = area.id;
+                   _.each(features, function(feature) {
+                     self.decorateFeature(feature, area.type, area.country);
+                   });
+
+                   // index by geoid
+                   _.each(features, function(feature) {
+                       featureMap[feature.properties.geoid] = feature;
+                   });
+
+                   if (counter === 0) {
```

Comment:
Minor nitpick maybe do the [decrement](https://github.com/CodeForAfrica/HURUmap-apps/pull/135/files#diff-28f11c7b5f6af8ab955cf76000148291R36) and check here so it not in many places when debugging.